ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -89,8 +89,6 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag]
     val model = H2OMOJOModel.createFromMojo(mojoData, Identifiable.randomUID(binaryModel._parms.algoName()))
 
     // pass some parameters set on algo to model
-    model.setFeaturesCols(binaryModel.modelDescriptor().features())
```

Comment:
features are already set inside createFromMojo and that is the only path how to create this spark model based on mojo

Reply:
On second thoughts, what if the testing (PROD) dataframe has columns named differently?

Reply:
I would not support that use-case. We would not now how to internally match the values. ( We use column name to fill the temporary data structure which is then passed to H2O).

We could use the order in which they came, but that is not really strong solution